### PR TITLE
DefaultFrozen Asset Accounting Fix

### PR DESCRIPTION
**Pull request by winder**
## Summary

Fix the asset default frozen cache so that asset holdings are created correctly.
Fix accounting issues related to re-opened asset holdings.

## Test Plan

New unit and integration tests.

### CODE REVIEW

File: idb/postgres/postgres_migrations.go [hunk 1]
```diff
@@ -651,6 +651,12 @@ func processAccountTransactions(txnrows <-chan idb.TxnRow, addressStr string, ad
 			}
 		}
 
+		if txn.AssetID == 82 && stxn.Txn.Sender.String() == "CM333ZN3KMASBRIP7N4QIN7AANVK7EJGNUQCNONGVVKURZIU2GG7XJIZ4Q" && stxn.Txn.Type == sdk_types.ApplicationCallTx {
```
Comment (brianolson) [inline idb/postgres/postgres_migrations.go:654]: debug code

File: importer/helper.go [hunk 2]
```diff
@@ -252,7 +252,9 @@ func updateAccounting(db idb.IndexerDb, genesisJSONPath string, numRoundsLimit i
 	}
 
 	lastlog := time.Now()
-	act := accounting.New()
+	cache, err := db.GetDefaultFrozen()
```
Comment (brianolson) [inline importer/helper.go:255]: if we keep a persistent default frozen map, accounting.go will update it on acfg

File: idb/postgres/postgres.go [hunk 3]
```diff
@@ -284,9 +289,9 @@ func (db *IndexerDb) GetAsset(assetid uint64) (asset types.AssetParams, err erro
 	return
 }
 
-// GetDefaultFrozen get {assetid:default frozen, ...} for all assets
+// GetDefaultFrozen get {assetid:default frozen, ...} for all assets, needed by accounting.
 func (db *IndexerDb) GetDefaultFrozen() (defaultFrozen map[uint64]bool, err error) {
-	rows, err := db.db.Query(`SELECT index, params -> 'df' FROM asset a`)
+	rows, err := db.db.Query(`SELECT index, coalesce((params ->> 'df')::boolean, false) FROM asset a`)
```
[the remark this answers is not in the document]
Comment (winder) [reply idb/postgres/postgres.go:294]: Cool trick, in that case I'd need to add `WHERE df is not NULL` to the query. I had some code to verify that the cache contained the key, but maybe that error checking is overkill.

File: importer/helper.go (hunk 2 above)
Comment (winder) [reply importer/helper.go:255]: Oops. Yeah I didn't mean to initialize this for every round.

File: idb/postgres/postgres.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (winder) [reply idb/postgres/postgres.go:294]: Looks like I got rid of that check anyway, so this should work nicely.

**Review by brianolson — APPROVED** [review]

**Pull request merged**